google-iam-credentials-v1/synth.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to synthesize generated parts of this library."""
+
+import synthtool as s
+import synthtool.gcp as gcp
+import synthtool.languages.ruby as ruby
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+gapic = gcp.GAPICMicrogenerator()
+library = gapic.ruby_library(
+    "iamcredentials", "v1",
+    proto_path="google/iam/credentials/v1",
+    extra_proto_files=[
+        "google/cloud/common_resources.proto",
+    ],
+    generator_args={
+        "ruby-cloud-gem-name": "google-iam-credentials-v1",
+        "ruby-cloud-title": "IAM Service Account Credentials V1",
+        "ruby-cloud-description": "The Service Account Credentials API creates short-lived credentials for Identity and Access Management (IAM) service accounts. You can also use this API to sign JSON Web Tokens (JWTs), as well as blobs of binary data that contain other types of tokens.",
+        "ruby-cloud-env-prefix": "IAM_CREDENTIALS",
+        "ruby-cloud-grpc-service-config": "google/iam/credentials/v1/iamcredentials_grpc_service_config.json",
+        "ruby-cloud-product-url": "https://cloud.google.com/iam",
```

Comment:
I looked for a better product page and found this: https://cloud.google.com/iam/docs/reference/credentials/rest

Unfortunately it's for the REST API.

Reply:
I did look at that, and decided against it because yes, it's for the REST API.